devito/ops/compiler.py

```diff
@@ -5,41 +5,33 @@
 from codepy.jit import compile_from_string
 from devito.logger import warning
 from devito.parameters import configuration
+from devito.archinfo import NVIDIAX
 
+__all__ = ['OPSCompiler']
 
-__all__ = ['CompilerOPS']
 
+class OPSMetaCompiler(configuration['compiler'].__class__):
 
-class CompilerOPS(configuration['compiler'].__class__):
     def __init__(self, *args, **kwargs):
         kwargs['cpp'] = True
+        super(OPSMetaCompiler, self).__init__(*args, **kwargs)
+
         self._ops_install_path = os.environ.get('OPS_INSTALL_PATH')
-        super(CompilerOPS, self).__init__(*args, **kwargs)
+        if not self._ops_install_path:
+            raise ValueError("Couldn't find OPS_INSTALL_PATH\
+                environment variable, please check your OPS installation")
 
-    def _cmdline(self, files, object=False):
-        if object:
-            ld_options = ['-c']
-            link = []
-        else:
-            ld_options = self.ldflags
-            link = ["-L%s" % ldir for ldir in self.library_dirs]
-            link.extend(["-l%s" % lib for lib in self.libraries])
-        return (
-            [self.cc]
-            + self.cflags
-            + ld_options
-            + ["-D%s" % define for define in self.defines]
-            + ["-U%s" % undefine for undefine in self.undefines]
-            + ["-I%s" % idir for idir in self.include_dirs]
-            + files
-            + link
-        )
+
+class OPSCompiler(OPSMetaCompiler):
+
+    def __init__(self, *args, **kwargs):
```

Comment:
this init is useless. It's just calling the superclass', which happens automatically if you don't override it